crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -274,20 +274,22 @@ where
     ) -> Result<Vec<Message>, ConversionError> {
         if topic.name.starts_with("tedge/alarms") {
             let mut mqtt_messages: Vec<Message> = Vec::new();
-            let topic_split: Vec<&str> = topic.name.split('/').collect();
-            if topic_split.len() == 5 {
-                let child_id = topic_split[4];
-                // Create a child device, if it does not exists already
-                if !child_id.is_empty() && !self.children.contains(child_id) {
-                    self.children.insert(child_id.to_string());
-                    mqtt_messages.push(Message::new(
-                        &Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC),
-                        format!("101,{child_id},{child_id},thin-edge.io-child"),
-                    ));
-                }
-            }
             self.size_threshold.validate(message)?;
             let mut messages = self.alarm_converter.try_convert_alarm(message)?;
+            if !messages.is_empty() {
```

Comment:
If the point is to have a behavior adapted to the `Syncing`/`Synced`  state of the converter (as stated in the presentation of this PR), I really don't like this `is_empty()` check, because it requires a deep knowledge of the `try_convert_alarm` method.

A better place to add the message to create the child device could be *inside* the `try_convert_alarm` method along the `Synced` match case.

Reply:
I did not get how does it will be different if I implement the `device create` message inside the `try_convert_alarm`? or outside.

Reply:
I have now a better understanding. This code is correct. The explanation linking the fix to the Syncing/Synced state of the converter was not, though. The correct explanation is aligned to with the code: one has to declare a child devices only if there are some messages to be sent on behalf of this child devices.

A comment might help.

```suggestion
           let mut messages = self.alarm_converter.try_convert_alarm(message)?;
           if !messages.is_empty() {
                // When there is some messages to be sent on behalf of a child device,
                // this child device must be declared first, if not done yet
```